Automattic/wp-calypso
# Me: Light styling for sidebar when collapsed

### Proposed Changes

1. Moves `client/my-sites/sidebar/collapse-sidebar.jsx` to `client/layout/sidebar/collapse-sidebar.jsx` for reuse between the two sidebars.
2. Adds `CollapseSidebar` toggle to `/me` sidebar.
3. Adds a bit of collapsed state styling to the `/me` sidebar.

#### Before

<img width="617" alt="image" src="https://user-images.githubusercontent.com/36432/171213188-a9907f62-da17-4b1e-a093-43ed2d598ac5.png">

#### After

<img width="548" alt="image" src="https://user-images.githubusercontent.com/36432/171719624-da4b35dd-4ca1-4df8-874c-618e28b0f512.png">

### Testing instructions

1. Navigate to http://calypso.localhost:3000/
4. Click on the `CollapseSidebar` toggle to collapse the left-hand sidebar.
5. Verify the sidebar is collapsed as expected.
6. Click on your avatar in the top right of the page.
7. Verify the new collapsed sidebar design on http://calypso.localhost:3000/me
8. Click on the `CollapseSidebar` toggle to expand the left-hand sidebar.
9. Verify the sidebar is expanded as expected.

Related to #62985
Previously #64199
